src/main/java/bro.java

```diff
@@ -0,0 +1,215 @@
+            System.out.println("No previous tasks found. Please create a new task list.");
+            return;
+        }
+
+        try (Scanner scanner = new Scanner(file)) {
+            while (scanner.hasNext()) {
+                String line = scanner.nextLine();
+                String[] parts = line.split(" \\| ");
+                switch(parts[0]) {
+                    case "T":
+                        Todo todo = new Todo(parts[2]);
+                        if (parts[1].equals("1")) {
+                            todo.markAsDone();
+                        }
+                        storer.add(todo);
+                        break;
+                    case "D":
+                        Deadline ddl = new Deadline(parts[2], parts[3]);
+                        if (parts[1].equals("1")) {
+                            ddl.markAsDone();
+                        }
+                        storer.add(ddl);
+                        break;
+                    case "E":
+                        Event event = new Event(parts[2], parts[3], parts[4]);
+                        if (parts[1].equals("1")) {
+                            event.markAsDone();
+                        }
+                        storer.add(event);
+                        break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Error while loading tasks: " + e.getMessage());
+        }
+    }
+
+    public static void echo() {
+
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            line = in.nextLine();
+
+            if (line.equals("Bye")) {
+                break;
+            }
+            System.out.println(line);
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
+
+    }
+
+    public static void addList() {
+
+        loadTasks();
+
+        String line;
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+            line = in.nextLine();
+
+            if (line.equals("Bye")) {
+                break;
+
+            } else if (line.equals("list")) {
+
+
+                for (int i = 0; i < storer.size(); i++) {
+                    System.out.println((i + 1) + ". " + storer.get(i));
+                }
+
+            } else if (line.startsWith("todo")) {
+
+                if (line.trim().length() <= 4) {
+                    System.out.println("Description of a todo cannot be empty. Please provide a task description.");
+                } else {
+                    String description = line.substring(5);
+                    Todo todo = new Todo(description);
+                    storer.add(todo);
+                    System.out.println("Got it. I've added this task\n " + todo);
+                    System.out.println("Now you have " + storer.size() + " tasks in the list.");
+                    saveTasks();
+                }
+
+            } else if (line.startsWith("deadline")) {
```

Comment:
The sections handling different task types (Todo, Deadline, Event) have similar patterns for adding and saving tasks. Can create a helper method to reduce duplication